lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -196,4 +247,99 @@ void SetMessageManagerHandler::initiate_release_session(
+  SessionSearchCriteria criteria(imsi, IMSI_AND_PDUID, pdu_id);
+  auto session_map = session_store_.read_sessions({imsi});
+  auto session_it  = session_store_.find_session(session_map, criteria);
+  auto& session    = **session_it;
+  if (!session_it) {
+    MLOG(MINFO) << " No session found for IMSI: " << imsi << " pdu id "
+                << pdu_id;
+    Status status(grpc::NOT_FOUND, "Sesion Not found");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  MLOG(MINFO) << "Received Session Notificaiton from UE with IMSI: " << imsi
+              << " of session Id " << session->get_session_id();
+  if (notif->common_context().sm_session_version() == 0) {
+    Status status(grpc::OUT_OF_RANGE, "Version number Out of Range");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  if (notif->rat_specific_notification().request_type() !=
+      EXISTING_PDU_SESSION) {
+    MLOG(MINFO) << " Wrong request type for sesion id"
+                << session->get_session_id();
+    Status status(grpc::UNKNOWN, "Unknown session request type");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  auto session_update = SessionStore::get_default_session_update(session_map);
+  auto session_id     = session->get_session_id();
+  SessionStateUpdateCriteria& session_crit = session_update[imsi][session_id];
+  m5g_enforcer_->m5g_state_change_action(
+		  imsi, session_crit, *notif, **session_it);
+  bool update_success = session_store_.update_sessions(session_update);
+  if (update_success) {
+     MLOG(MINFO) << "Successfully updated SessionStore "
+		  << "of subscriber" << imsi;
+     response_callback(Status::OK, SmContextVoid());
+     return;
+  }
+  Status status(grpc::ABORTED, "update operation aborted in  middle");
+  response_callback(status, SmContextVoid());
+  return;
+}
+void SetMessageManagerHandler::idle_mode_change_sessions_handle(
+    const SetSmNotificationContext* notif,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  // extract IMSI value from proto
+  auto imsi = notif->common_context().sid().id();
+  if (notif->rat_specific_notification().request_type() !=
+      EXISTING_PDU_SESSION) {
+    MLOG(MINFO) << " Wrong request type for imsi" << imsi;
+    Status status(grpc::UNKNOWN, "Unknown session request type");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  if (notif->common_context().sm_session_version() == 0) {
+    Status status(grpc::OUT_OF_RANGE, "Version number Out of Range");
+    response_callback(status, SmContextVoid());
+    return;
+  }
+  auto session_map = session_store_.read_sessions({imsi});
+  int count        = 0;
+  for (auto& session : session_map[imsi]) {
+    auto session_update = SessionStore::get_default_session_update(session_map);
+    auto session_id     = session->get_session_id();
+    SessionStateUpdateCriteria& session_crit = session_update[imsi][session_id];

```

Comment:
lets use something like `sessionState_uc` instead of `session_crit`

Reply:
Fixed